NDTensors/src/lib/UnallocatedArrays/src/unallocatedfill.jl

```diff
@@ -0,0 +1,87 @@
+using FillArrays: FillArrays, AbstractFill, Fill, broadcasted_fill, kron_fill, mult_fill
+using NDTensors.SetParameters: Position, set_parameters
+import Base: +
+
+struct UnallocatedFill{ElT,N,Axes,Alloc} <: AbstractFill{ElT,N,Axes}
+  f::Fill{ElT,N,Axes}
+  alloc::Alloc
+
+  function UnallocatedFill{ElT,N,Axes}(
```

Comment:
@mtfishman By putting this in the struct I eliminate the possibility for someone to put any arbitrary thing for `Alloc`. That way it will throw an error if Alloc isn't an AbstractArray type with the correct `eltype` and `N`. Then when we have some other concrete example, like `heterogenouscomputing.jl` we can simply add another constructor to the struct which takes that object and appropriately makes the UnallocatedArray

Reply:
Seems fine for now.

Reply:
@mtfishman I am moving this constructor and the one in UnallocatedZeros outside of the struct